packages/design-toolkit/src/components/flashcard/flashcard.test.tsx

```diff
@@ -0,0 +1,104 @@
+ * governing permissions and limitations under the License.
+ */
+
+import { uuid } from '@accelint/core';
+import '@testing-library/jest-dom';
+import { render, screen } from '@testing-library/react';
+import { Fragment } from 'react/jsx-runtime';
+import { describe, expect, it } from 'vitest';
+import {
+  Flashcard,
+  FlashcardAdditionalData,
+  FlashcardDetailLabel,
+  FlashcardDetailsList,
+  FlashcardDetailValue,
+  FlashcardHeader,
+  FlashcardHero,
+  FlashcardSubheader,
+} from '.';
+
+function setup(isLoading = false) {
+  const details = [
+    { label: 'OBJECTID', value: 75, id: uuid() },
+    { label: 'NAME', value: 'Acrisure Stadium', id: uuid() },
+    { label: 'LATITUDE', value: '40-26-48.22N', id: uuid() },
+    { label: 'LONGITUDE', value: '080-00-56.66W', id: uuid() },
+    { label: 'CITY', value: 'Pittsburgh', id: uuid() },
+    { label: 'STATE', value: 'PA', id: uuid() },
+    { label: 'STATUS_CODE', value: 'Open', id: uuid() },
+    { label: 'OPENING_ON', value: null, id: uuid() },
+  ];
+
+  render(
+    <Flashcard isLoading={isLoading}>
+      <FlashcardHero data-testid='hero'>
+        <FlashcardHeader>IDENTIFIER</FlashcardHeader>
+        <FlashcardSubheader>DATA</FlashcardSubheader>
+      </FlashcardHero>
+      <FlashcardAdditionalData>SECONDARY_DATA_01</FlashcardAdditionalData>
+      <FlashcardAdditionalData>SECONDARY_DATA_02</FlashcardAdditionalData>
+      <FlashcardDetailsList data-testid='secondary'>
+        {details.map((detail) => (
+          <Fragment key={detail.id}>
+            <FlashcardDetailLabel>{detail.label}</FlashcardDetailLabel>
+            <FlashcardDetailValue>{detail.value}</FlashcardDetailValue>
+          </Fragment>
+        ))}
+      </FlashcardDetailsList>
+    </Flashcard>,
+  );
+}
+
+describe('Flashcard', () => {
+  it('should render', () => {
+    setup();
+    expect(screen.getByText('IDENTIFIER')).toBeDefined();
+    expect(screen.getByText('DATA')).toBeDefined();
+    expect(screen.getByText('SECONDARY_DATA_01')).toBeDefined();
+    expect(screen.getByText('SECONDARY_DATA_02')).toBeDefined();
+  });
+
+  it('should only show 5 additional details', () => {
+    setup();
+
+    // First item to be hidden by selector nth-of-type
+    const detailEntries = document.getElementsByClassName(
+      'nth-of-type-[n+6]:hidden',
+    );
+
+    // 2 sets of 8
+    expect(detailEntries.length).toEqual(16);
+
+    // First items should be visible.
+    expect(detailEntries[0]).toBeVisible();
+    // TODO: Testing for visibility is wonky.
+    // expect(detailEntries[11]).not.toBeVisible();
+  });
+
+  it('should not show secondary data field while loading', () => {
+    setup(true);
+
+    // Should not render header text.
+    const header = screen.queryByText('IDENTIFIER');
+    expect(header).toBeNull();
+
+    const hero = screen.getByTestId('hero');
+    expect(hero).toBeDefined();
+
+    // Two skellington components
+    expect(hero.childElementCount).toEqual(2);
```

Comment:
there should be some kind of grabbable prop on the skeleton components I think to assert they're specifically what the children are:
`screen.getByTestId()` is what I would grab, and I would add a data-testid attribute to the skeleton components if they dont' have any identifiable other attributes in the screen.debug() you could grab.